### PR TITLE
Exclude example and integration tests from coverage

### LINKED ISSUE #423: Run a coverage report without integration tests
The coverage tests report too high a coverage value because the integration tests are inflating the value and hiding the actual coverage of unit tests.

### PR DESCRIPTION
**Pull request by dorukozturk**
## Motivation

Fixes #423 
Coverage is only useful in the context of unit testing. Integration tests create the illusion of a higher coverage than what we currently have. Example tests are just executing tutorial code but they are not asserting any conditions explicitly so I excluded also the example tests from coverage. 

## How to test the behavior?
```
make coverage
```

## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -47,7 +47,7 @@ commands = {[testenv:build]commands}
 # Envs that will only be executed on CI that does coverage reporting
 [testenv:coverage]
 commands =
-    coverage run test.py
+    coverage run test.py -f -p
```
Comment (jcfr) [inline tox.ini]: I suggest to use the long argument ``--form``  and ``--pynwb``

Comment (dorukozturk) [reply tox.ini]: @jcfr I changed it to use the long argument names. Is it good now?

**Comment by dorukozturk** [comment]
@ajtritt Jc is at vacation and might not see this. Can you review it?

**Review by ajtritt — APPROVED** [review]

**Comment by waxlamp** [comment]
@dorukozturk, will this trigger an updated coverage report? Where can we see that? Thanks for this PR!

**Comment by dorukozturk** [comment]
@waxlamp Yes, an update already happened in the report. Coverage was %77 before and now it is %68 due to the fact that we excluded example and integration tests. [This](https://codecov.io/gh/NeurodataWithoutBorders/pynwb/tree/dev/src/pynwb) is the current coverage.

**Comment by waxlamp** [comment]
Awesome!

Should we exclude `legacy` from coverage as well, or is the testing there needed?

**Comment by dorukozturk** [comment]
I think we want to keep it and even improve testing. @nicain Is that the case?

**Pull request merged**